cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -703,6 +703,11 @@ TEST(Substrait, ExtensionSetFromPlan) {
         "extension_uri_anchor": 7,
         "uri": ")" + substrait::default_extension_types_uri() +
                                R"("
+      },
+      {
+        "extension_uri_anchor": 18,
+        "uri": ")" + std::string(kSubstraitArithmeticFunctionsUri) +
```

Comment:
I think it would be useful, at least for PyArrow's sake, to expose the string value of this constant (and similar ones) like `default_extension_types_uri()` does.

Reply:
I converted these constants to std::string.

Reply:
I'm not sure if that will make them available to python or not without the functions though.  I'll try that out later.

Reply:
I considered adding these functions but, at the end of the day, this is information that the caller should be getting from Substrait and I don't know if Arrow needs to provide them.  I don't want to hold up this PR on this so we still think these functions are needed we can create a follow-up PR and have the discussion there.

Reply:
Actually, I had to revert these back to `char *` to make the linter happy.  My thought  on a follow-up PR still holds I think.